packages/next/src/server/image-optimizer.ts

```diff
@@ -44,23 +39,26 @@ const VECTOR_TYPES = [SVG]
 const BLUR_IMG_SIZE = 8 // should match `next-image-loader`
 const BLUR_QUALITY = 70 // should match `next-image-loader`
 
-let sharp: typeof import('sharp') | undefined
+let sharp: typeof import('sharp')
 
 try {
-  sharp = require(process.env.NEXT_SHARP_PATH || 'sharp')
+  sharp = require('sharp')
   if (sharp && sharp.concurrency() > 1) {
     // Reducing concurrency should reduce the memory usage too.
     // We more aggressively reduce in dev but also reduce in prod.
     // https://sharp.pixelplumbing.com/api-utility#concurrency
     const divisor = process.env.NODE_ENV === 'development' ? 4 : 2
     sharp.concurrency(Math.floor(Math.max(cpus().length / divisor, 1)))
   }
-} catch (e) {
-  // Sharp not present on the server, Squoosh fallback will be used
+} catch (e: unknown) {
+  if (isError(e) && e.code === 'MODULE_NOT_FOUND') {
+    throw new Error(
+      'Module `sharp` not found. Please run `npm install --cpu=wasm32 sharp` to install it.'
```

Comment:
Why are we recommending `wasm32` here, couldn't another platform be supported and they it hasn't been installed yet?

Also what do we think about auto-installing for the user similar to our `verifyTypeScriptSetup` handling?

Reply:
Since my PR added `sharp` to `packages/next/package.json`, it means that the only time we will failing during `require('sharp')` is the time when it failed to install. The only time it will fail to install is if there is not matching prebuilt binary and it can't build from source. In that case, it seems its likely the current platform would support wasm, so we print this helpful error message.

I don't think typescript has native vs wasm variants so that use case is a little different. We also don't list typescript under `optionalDependencies`.

Reply:
I took a look at `verifyTypeScriptSetup` and I think we could follow a similar pattern to automatically install https://github.com/vercel/next.js/blob/daf891262720d8430a2c4e1e50017c8d9256d738/packages/next/src/lib/install-dependencies.ts#L18

But we don't know for sure if installing wasm will work so we might get into an infinite loop.

I checked with the two big providers, [Codesandbox](https://vercel.slack.com/archives/CL4LV7THD/p1693387243290309?thread_ts=1693339993.221929&cid=CL4LV7THD) and [Stackblitz](https://vercel.slack.com/archives/C01PNUXPB8Q/p1707241368512289?thread_ts=1707239376.935939&cid=C01PNUXPB8Q), and both indicate that this should work automatically without the user manually installing wasm variant.